src/runtime/story.rs

```diff
@@ -0,0 +1,192 @@
+use std::{
+    borrow::Borrow,
+    fmt::{self, Write},
+    rc::Rc,
+};
+
+use crate::runtime_graph::RuntimeGraph;
+
+use super::{
+    container::Container,
+    control_command::ControlCommand,
+    value::Value,
+    variable::{ReadCount, VariableReference},
+    RuntimeObject,
+};
+
+#[derive(Debug)]
+pub struct Story<Output> {
+    output_text: Output,
+    glue: bool,
+    cursors: Vec<(Rc<Container>, usize)>,
+}
+
+impl<Output> Story<Output> {
+    fn new<Graph>(graph: Graph, output_text: Output) -> Self
+    where
+        Graph: Borrow<RuntimeGraph>,
+    {
+        let cursors = vec![(graph.borrow().root_container.clone(), 0)];
+        Self {
+            output_text,
+            glue: false,
+            cursors,
+        }
+    }
+
+    pub fn step(&mut self) -> Result<(), ()>
+    where
```

Comment:
I would prefer to put the `where Output: Write` on the `impl` as a whole rather than have it per function